Time to update to Go Mod 1.15?

I was building the provider image with the provided Dockerfile and I was getting this error:

```
 => ERROR [builder 9/9] RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 GO111MODULE=on go build -a -o provider cmd/provider/main.go                                                                       26.6s
------
 > [builder 9/9] RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 GO111MODULE=on go build -a -o provider cmd/provider/main.go:
#17 26.34 # sigs.k8s.io/controller-runtime/pkg/manager
#17 26.34 /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.8.0/pkg/manager/manager.go:53:2: duplicate method Start
#17 26.34 note: module requires Go 1.15

```

I've tried upgrading to Go Mod 1.15 and it works.. so I will send a PR with the changes I made.. but someone needs to validate that something else is not broken because of that..

--- a/cluster/Dockerfile
+++ b/cluster/Dockerfile
@@ -1,5 +1,5 @@
 # Build the manager binary
-FROM golang:1.13 as builder
+FROM golang:1.16 as builder
 
 WORKDIR /workspace
 # Copy the Go Modules manifests

--- a/go.mod
+++ b/go.mod
@@ -1,6 +1,6 @@
 module github.com/crossplane/provider-template
 
-go 1.13
+go 1.16
 
 require (
 	github.com/crossplane/crossplane-runtime v0.13.0

